simulated_data_sources.go

```diff
@@ -183,9 +181,14 @@ func (sps *SimPulseSource) blockingRead() error {
 		for i := 0; i < sps.cycleLen; i++ {
 			datacopy[i] += RawType(rand.Intn(21) - 10)
 		}
-		seg := DataSegment{rawData: datacopy, framesPerSample: 1}
+		seg := DataSegment{
+			rawData:         datacopy,
+			framesPerSample: 1,
+			firstFramenum:   sps.lastFrameNum,
```

Comment:
This reads wrong? Shouldn't it be something like `lastFrameNum+1`? I'm guessing the answer is not, but a comment here explaining it would be nice.

Reply:
Can do. I'll think about a comment and whether the variable name is accurate.